state.go

```diff
@@ -21,6 +21,61 @@ import (
 	"github.com/pkg/errors"
 )
 
+// State transitions:
+//
+//                             init
+//
+//                               |        (wait timeout expired)
+//                               |   +---------------------------------+
+//                               |   |                                 |
+//                               v   v                                 |
+//                                         (auth req. failed)
+//                          bootstrapped ----------------------> authorize wait
+//
+//                                |
+//                                |
+//                                |  (auth data avail.)
+//                                |
+//                                v
+//
+//                           authorized
+//
+//          (update needs     |   |
+//           commit)          |   |
+//         +------------------+   |
+//         |                      |
+//         v                      |          (wait timeout expired)
+//                                |    +-----------------------------+
+//   update commit                |    |                             |
+//                                v    v                             |
+//         |                                (no update)
+//         +---------------> update check ---------------->  update check wait
+//
+//                                |
+//                                | (update ready)
+//                                v
+//
+//                           update fetch
+//
+//                                |
+//                                | (update fetched)
+//                                v
+//
+//                          update install
+//
+//                                |
+//                                | (update installed,
+//                                |  enabled)
+//                                |
+//                                v
+//
+//                              reboot
+//
+//                                |
+//                                v
+//
+//                              final (daemon exit)
```

Comment:
We should be, but we are not at the moment. It should be possible to store the ID of new image for later comparison to detect if mender was started in the same image or a new one. Preferably, this should be a part of update state information we want to preserver across reboots.

FWIW we have `Restart=on-abort` in service definition in `meta-mender`, so unless we crash systemd will not restart the process after exiting.